lsp/ty.lua

```diff
@@ -0,0 +1,37 @@
+---@brief
+---
+--- https://github.com/astral-sh/ty
+---
+--- `ty`, a static type checker and language server for python.
+---
+--- The language server can be installed with `uv`:
+--- ```sh
+--- uv tool install ty
+--- ```
+
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'ty', 'server' },
+    filetypes = { 'python' },
+    root_dir = function(fname)
+      return util.root_pattern(
+        'ty.toml',
+        'pyproject.toml',
+        'setup.py',
+        'setup.cfg',
+        'requirements.txt',
+        'Pipfile',
+        '.git'
+      )(fname)
+    end,
+  },
+  docs = {
+    description = [[
+https://github.com/astral-sh/ty
+`ty`, a static type checker and language server for python.
+The language server can be installed with `uv`: `uv tool install ty`
+]],
+  },
+}
```

Comment:
This isn't correct. This is what the legacy looks like.

```suggestion
return {
  cmd = { 'ty', 'server' },
  filetypes = { 'python' },
  root_markers = {
    'ty.toml',
    'pyproject.toml',
    'setup.py',
    'setup.cfg',
    'requirements.txt',
    'Pipfile',
    '.git'
  },
}
```